client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-other-platform-detected-import/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { StepContainer } from '@automattic/onboarding';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPlatformImporterName } from 'calypso/blocks/import/util';
+import DocumentHead from 'calypso/components/data/document-head';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import { useAnalyzeUrlQuery } from 'calypso/data/site-profiler/use-analyze-url-query';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { ImporterPlatform } from 'calypso/lib/importer/types';
+import { Step } from '../../types';
+import { ImportPlatformForwarder } from './components/importer-forwarding-details';
+import './style.scss';
+
+export const Scanning = () => {
+	return (
+		<div className="site-migration-other-platform__scanning">
+			<LoadingEllipsis />
+		</div>
+	);
+};
+
+const SiteMigrationOtherPlatform: Step = function ( { navigation } ) {
+	const translate = useTranslate();
+	const [ query ] = useSearchParams();
+	const from = query.get( 'from' ) as string;
+
+	const [ platform, setPlatform ] = useState< ImporterPlatform | null >(
+		query.get( 'platform' ) as ImporterPlatform | null
+	);
+
+	const shouldAnalyzeUrl = platform === null;
+
+	const {
+		isFetching: isAnalyzingUrl,
+		data: siteInfo,
+		isSuccess,
+	} = useAnalyzeUrlQuery( from, shouldAnalyzeUrl );
```

Comment:
It only runs the site analyzes when the platform is not available on the URL.

It is necessary because when the user uses the back button on some importing page, we lose the platform name. 
So, instead of updating all importer options (WPCOM and WPorg), I just made the step analysis again.